rootfs/deploy.py

```diff
@@ -96,6 +96,14 @@ def download_file(tar_path):
 with tarfile.open("apptar", "r:gz") as tar:
     tar.extractall("/app/")
 log("extracting tar file complete")
+buildargs = json.loads(os.getenv('DOCKER_BUILD_ARGS', '{}'))
+# inject docker build args into the Dockerfile so we get around Dockerfiles that don't have things
+# like PORT defined.
+with open("/app/Dockerfile", "a") as dockerfile:
+    # ensure we are on a new line
+    dockerfile.write("\n")
+    for envvar in buildargs:
+        dockerfile.write("ARG {}\n".format(envvar))
```

Comment:
I think the `with` block can be limited to `float(client.version()['ApiVersion']) < 1.13` because the error was changed to a warning in docker/docker#26249 which is part of the 1.13.0 milestone (unless you also want to suppress the warning).